avm-transpiler/src/transpile_contract.rs

```diff
@@ -91,8 +91,9 @@ impl From<CompiledAcirContractArtifact> for TranspiledContractArtifact {
         let mut functions: Vec<AvmOrAcirContractFunctionArtifact> = Vec::new();
 
         for function in contract.functions {
-            // TODO(4269): once functions are tagged for transpilation to AVM, check tag
-            if function.custom_attributes.contains(&"public".to_string()) {
+            if function.custom_attributes.contains(&"public".to_string())
+                || function.name == "public_dispatch"
+            {
```

Comment:
Because otherwise nothing would work XD. This PR only adds a public_dispatch function, but nothing is using it yet. As we discussed on Slack, we can see what to do with the other functions once they are not deployed.